torchvision/datasets/cub200.py

```diff
@@ -0,0 +1,195 @@
+            if train_test_list[i, 1] == 1 and train:
+                imgs.append((full_path, int(fname[0:3]) - 1))
+            elif train_test_list[i, 1] == 0 and not train:
+                imgs.append((full_path, int(fname[0:3]) - 1))
+            if os.path.split(fname)[0][4:] not in classes:
+                classes.append(os.path.split(fname)[0][4:])
+                class_to_idx.append(int(fname[0:3]) - 1)
+
+        return imgs, classes, class_to_idx
+
+
+class CUB200(data.Dataset):
+    """`CUB200 <http://www.vision.caltech.edu/visipedia/CUB-200.html>`_ Dataset.
+       `CUB200 <http://www.vision.caltech.edu/visipedia/CUB-200-2011.html>`_ Dataset.
+
+    Args:
+        root (string): Root directory of dataset the images and corresponding lists exist
+            inside raw folder
+        train (bool, optional): If True, creates dataset from ``training.pt``,
+            otherwise from ``test.pt``.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        year (int): Year/version of the dataset. Available options are 2010 and 2011
+    """
+    urls = []
+    raw_folder = 'raw'
+
+    def __init__(self, root, year, train=True, transform=None, target_transform=None, download=False,
+                 loader=default_loader):
+        self.root = os.path.expanduser(root)
+        self.transform = transform
+        self.target_transform = target_transform
+        self.train = train
+        self.year = year
+        self.loader = loader
+
+        assert year == 2010 or year == 2011, "Invalid version of CUB200 dataset"
+        if year == 2010:
+            self.urls = ['http://www.vision.caltech.edu/visipedia-data/CUB-200/images.tgz',
+                         'http://www.vision.caltech.edu/visipedia-data/CUB-200/lists.tgz']
+
+        elif year == 2011:
+            self.urls = ['http://www.vision.caltech.edu/visipedia-data/CUB-200-2011/CUB_200_2011.tgz']
+
+        if download:
+            self.download()
+
+        if not self._check_exists():
+            raise RuntimeError('Dataset not found. You can use download=True to download it')
+
+        self.imgs, self.classes, self.class_to_idx = build_set(os.path.join(self.root, self.raw_folder),
+                                                               self.year, self.train)
+
+    def __getitem__(self, index):
+        """
+        Args:
+            index (int): Index
+
+        Returns:
+            tuple: (image, target) where target is index of the target class.
+        """
+        path, target = self.data_set[index]
```

Comment:
`data_set` seems not to be defined?

Reply:
This is fixed in the latest commit. Thank you for pointing that out.